File: randpi.cpp
#include<bits/stdc++.h>
using namespace std;

int main()
{
  double inside = 0;
  double total = 1000000;
  int l = 8000;

  for(int i = 0; i<total ; i++)
  {
      double x = rand()%(l+1) - l/2;
      double y = rand()%(l+1)- l/2;
      double dist = sqrt(x*x + y*y);
      cout<<x<<" "<<y<<" "<<dist<<endl;
      if(dist<=l/2) inside++;
  }
  cout<<4*(inside/total);
}

File: randpi.py
import math

inside = 0

for i in range(0,10000) :
	randx = randint(0,4) - 2
	randy = randint(0,4) - 2
    
    ans = math.pow(randx,2)
    ans = ans + randy*randy

    dist = math.sqrt(ans)

    if dist<=2 :
    	inside++

rat = inside/10000    	

rat = 4*rat
print(rat)


